cmd/basin/publication.go

```diff
@@ -330,6 +332,92 @@ func newPublicationListCommand() *cli.Command {
 	}
 }
 
+func newPublicationDealsCommand() *cli.Command {
+	var publication, provider string
+	var limit, latest int
+	var offset int64
+
+	return &cli.Command{
+		Name:  "deals",
+		Usage: "list deals of a given publications",
+		Flags: []cli.Flag{
+			&cli.StringFlag{
+				Name:        "publication",
+				Usage:       "Publication name",
+				Destination: &publication,
+				Required:    true,
+			},
+			&cli.StringFlag{
+				Name:        "provider",
+				Usage:       "The provider's address and port (e.g. localhost:8080)",
+				Destination: &provider,
+				Value:       DefaultProviderHost,
+			},
+			&cli.IntFlag{
+				Name:        "limit",
+				Usage:       "The number of deals to fetch",
+				Destination: &limit,
+				Value:       10,
+			},
+			&cli.IntFlag{
+				Name:        "latest",
+				Usage:       "The latest N deals to fetch",
+				Destination: &latest,
+				Value:       10,
+			},
+			&cli.Int64Flag{
+				Name:        "offset",
+				Usage:       "The epoch to start from",
+				Destination: &offset,
+				Value:       0,
+			},
+		},
+		Action: func(cCtx *cli.Context) error {
+			ns, rel, err := parsePublicationName(publication)
+			if err != nil {
+				return err
+			}
+
+			bp, err := basinprovider.New(cCtx.Context, provider)
+			if err != nil {
+				return fmt.Errorf("new basin provider: %s", err)
+			}
+			defer bp.Close()
+
+			var deals []app.DealInfo
+			if latest > 0 {
```

Comment:
depending if latest is set or not calls one method `LatestDeals` or the other `Deals`